src/Features/Core/Portable/QuickInfo/QuickInfoItem.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    public sealed class QuickInfoItem
+    {
+        /// <summary>
+        /// The span of the document that the item is based on.
+        /// </summary>
+        public TextSpan Span { get; }
+
+        /// <summary>
+        /// Descriptive tags from the <see cref="Completion.CompletionTags"/> type.
+        /// These tags may influence how the item is displayed.
+        /// </summary>
+        public ImmutableArray<string> Tags { get; }
+
+        /// <summary>
+        /// One or more <see cref="QuickInfoSection"/> describing the item.
+        /// </summary>
+        public ImmutableArray<QuickInfoSection> Sections { get; }
+
+        /// <summary>
+        /// Alternate regions of the document that help describe the item.
+        /// </summary>
+        public ImmutableArray<TextSpan> RelatedSpans { get; }
+
+        private QuickInfoItem(
+            TextSpan span,
+            ImmutableArray<string> tags,
+            ImmutableArray<QuickInfoSection> sections,
+            ImmutableArray<TextSpan> relatedSpans)
+        {
+            this.Span = span;
+            this.Tags = tags.IsDefault ? ImmutableArray<string>.Empty : tags;
+            this.Sections = sections.IsDefault ? ImmutableArray<QuickInfoSection>.Empty : sections;
+            this.RelatedSpans = relatedSpans.IsDefault ? ImmutableArray<TextSpan>.Empty : relatedSpans;
+        }
+
+        public static QuickInfoItem Create(
+            TextSpan span,
+            ImmutableArray<string> tags = default,
+            ImmutableArray<QuickInfoSection> sections = default,
+            ImmutableArray<TextSpan> relatedSpans = default)
+        {
+            return new QuickInfoItem(span, tags, sections, relatedSpans);
+        }
+
+        public bool IsEmpty
+        {
+            get
+            {
+                return this == Empty
+                    || (this.Span == default
+                    && this.Tags.Length == 0
+                    && this.Sections.Length == 0
+                    && this.RelatedSpans.Length == 0);
+            }
+        }
+
+        public static readonly QuickInfoItem Empty = Create(default);
```

Comment:
What's the meaning of Empty vs. null, given this is a class?

Reply:
It's consistent with the Completion API, where `CompletionList` also has an empty property. I think the idea in Completion was originally that `CompletionService.GetCompletionsAsync(...)` would never return null. However, looking at the code, it doesn't see that's actually true. I think we can remove it from Quick Info.